Lib/typing.py

```diff
@@ -202,6 +202,20 @@ def _check_generic(cls, parameters, elen):
                         f" actual {alen}, expected {elen}")
 
 
+def _deduplicate(params):
+    # Weed out strict duplicates, preserving the first of each occurrence.
+    all_params = set(params)
```

Comment:
This method is used by `Union` and for `Union` all parameters are hashable, so it will affect the performance of `Union`.

In case if we will support mutable values we should create a separate `_deduplicate` function for `Literal` type.

Reply:
You're right, didn't see that in the last commit you separated out `_dedup` and re-used for both `Union` and `Literal`. Yes in that scenario we can create a separate function.